### PR TITLE
PP-1216: TestReservations.test_sched_cycle_starts_on_resv_end test fa…

### PR DESCRIPTION
**Pull request by shilpakodli**
…iled with race condition

#### Bug/feature Description
* *Clearly and concisely describe the problem/feature as it affects the customer (this cannot be empty)*
* *Bugs: clearly describe the steps to easily reproduce the problem; you can include short illustrative excerpts from log files or terminal sessions*
* *Bugs: if the issue is intermittent and not easily reproducible, describe the situation/scenarios/configurations where it was likely triggered*
* *If there is an issue ID, link to it: [PP-1216](https://pbspro.atlassian.net/browse/PP-1216)*

#### Cause / Analysis / Design
* *According to the test case reservation should start after CT+5 sec. , but reservation is started immediately after submitting the reservation. Hence test case is failed. 
Scenario was:
saving current time.
setting ncpus values:2
after that reservation is submitted with current time + 5 sec. But setting ncpus value takes some time(near about 5 sec. ) which result into reservation is started immediately after submitting. Test case checking for reservation confirmation after submission but at the same time reservation is in running state.

#### Solution Description
CT is saved right before reservation submission and increased start and end Time as CT+10 and CT+30 respectively. Also checking for RESV_RUNNING state after job submission.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/functional/pbs_reservations.py [hunk 1]
```diff
@@ -203,7 +203,11 @@ def test_sched_cycle_starts_on_resv_end(self):
...
             jid + ";" + msg,
             max_attempts=30)
+        
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|2')}
+        self.server.expect(RESV, a, rid)
+        
         self.server.log_match(
             rid.split('.')[0] + ";deleted at request of pbs_server",
-            id=rid.split('.')[0], max_attempts=30)
+            id=rid.split('.')[0], interval=5)
```
Comment (bhroam) [inline test/tests/functional/pbs_reservations.py]: You do the rid.split() twice in this log match.  Do it outside and save the value into a variable so you only need to do it once.

**Comment by bhroam** [comment]
You have some trailing whitespace.  This will fail the pep8 check.

**Comment by anamikau** [comment]
@shilpakodli codacy is still compaining about trailing whitespaces. Please run pep8 and fix those.

**Comment by shilpakodli** [comment]
@anamikau  I have corrected the code and pushed my changes also I have addressed Bhroam's review comments.

**Comment by bhroam** [comment]
Please squash and rebase for the merge.

**Comment by shilpakodli** [comment]
I have squashed my commits and rebased my branch. Please take this PR forward.

**Comment by bhroam** [comment]
I don't see evidence of testing.  Can you run the test and add it to the PR?

**Comment by shilpakodli** [comment]
@bhroam I have attatched test logs to the PP ticket. I had run the test and attached test logs to the Internal Jira ticket.

**Comment by shilpakodli** [comment]
@PBSPro/pbspro-maintainers : Please take this PR forward.

**Comment by bhroam** [comment]
Your PR is out of date, please rebase.

**Comment by bhroam** [comment]
I merged in the PR.  @shilpakodli as a note for the future, please attach testing logs to the PR itself, not the PP ticket.

**Pull request merged**